compiler/src/iree/compiler/Codegen/Common/DecomposeBatchMmt4DOps.cpp

```diff
@@ -130,6 +132,35 @@ void DecomposeBatchMmt4DOpsPass::runOnOperation() {
   MLIRContext *ctx = &getContext();
   auto funcOp = getOperation();
 
+  funcOp.walk([&](linalg::BatchMmt4DOp op) {
+    auto out = op.getDpsInitOperand(0)->get();
+    auto outType = out.getType().cast<RankedTensorType>();
+    // Tile only non unit batch dimensions with tile size equals to 1.
+    if (outType.getShape()[0] <= 1) {
+      return;
+    }
+    SmallVector<int64_t> tileSizes{1};
+    auto tilingInterfaceOp = cast<TilingInterface>(op.getOperation());
+    scf::SCFTileAndFuseOptions options{};
+    options.setTilingOptions(scf::SCFTilingOptions().setTileSizes(
+        getAsIndexOpFoldResult(ctx, tileSizes)));
+
+    IRRewriter rewriter(ctx);
+    FailureOr<scf::SCFTileAndFuseResult> tiledResults =
+        scf::tileConsumerAndFuseProducerGreedilyUsingSCFForOp(
+            rewriter, tilingInterfaceOp, options);
+    if (failed(tiledResults)) {
+      return;
+    }
+
+    SmallVector<Value> replacements;
+    replacements.resize(op.getNumResults());
+    for (const auto &[index, result] : llvm::enumerate(op.getResults())) {
+      replacements[index] = tiledResults->replacements.lookup(result);
+    }
+    rewriter.replaceOp(op, replacements);
```

Comment:
We can simplify it to `rewriter.replaceOp(op, tiledResults->replacements[op.getResult()]);`.